integration/docker/privileges_test.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package docker
+
+import (
+	"io/ioutil"
+	"os"
+
+	. "github.com/clearcontainers/tests"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("docker privileges", func() {
+	var (
+		args		[]string
+		id		string
+		secondId	string
+		thirdId		string
+		testImage	string
+		exitCode	int
+	)
+
+	AfterEach(func() {
+		Expect(RemoveDockerContainer(id)).To(BeTrue())
+		Expect(ExistDockerContainer(id)).NotTo(BeTrue())
+		Expect(RemoveDockerContainer(secondId)).To(BeTrue())
+		Expect(ExistDockerContainer(secondId)).NotTo(BeTrue())
+		_, _, exitCode := DockerRmi(testImage)
+		Expect(exitCode).To(Equal(0))
+	})
+
+	Context("check no-new-privileges flag", func() {
+		It("should display the correct effective uid", func() {
+			Skip("Issue https://github.com/clearcontainers/runtime/issues/880")
+			id = randomDockerName()
+			args = []string{"-d", "--name", id, GccImage, "bash", "-c", "sleep 30"}
+			_, _, exitCode = DockerRun(args...)
+			Expect(exitCode).To(Equal(0))
+			args = []string{id, "bash", "-c", "echo -e '#include<stdio.h>\n#include <unistd.h>\n#include <sys/types.h>\nint main (int argc, char *argv[])\n{printf(\"Effective uid: %d\", geteuid());return 0;}' > demo.c && make demo"}
```

Comment:
I'm not really clear why it's necessary to build our own binary. Can't we just use a standard image with a standard version of `/bin/id` (which shows the EUID) and just create a container, copy `/bin/id` to `/bin/id-foo` or something, `chmod -s` it, commit the image, and re-run the test using that new image and passing `--security-opt=no-new-privileges`?